### PR TITLE
Set Rolebinding OwnerReference  

### PR DESCRIPTION
**Pull request by suaas21**
xref: #672

Tasks:
 - [x] set workload as rolebinding  owner

### CODE REVIEW

File: pkg/controller/rbac_jobs.go [hunk 1]
```diff
@@ -86,9 +87,18 @@ func (c *StashController) ensureCronJobRoleBinding(resource *core.ObjectReferenc
 		Namespace: resource.Namespace,
 	}
 
+	bcObj, err := c.stashClient.StashV1beta1().BackupConfigurations(resource.Namespace).Get(resource.Name, metav1.GetOptions{})
```
Comment (hossainemruz) [inline pkg/controller/rbac_jobs.go]: We need to set workload as owner for `BackupSidecarRoleBinding`, `RestoreInitContainerRoleBinding`, `BackupJobRoleBinding`.

Not `CronJobRoleBinding`. CronJob has no relation with workload.

File: pkg/controller/rbac_jobs.go [hunk 2]
```diff
@@ -86,9 +87,18 @@ func (c *StashController) ensureCronJobRoleBinding(resource *core.ObjectReferenc
...
+	if err != nil {
+		return err
+	}
+
+	exists, objRef := c.checkAndGetWorrkloadReference(bcObj)
+
 	// ensure role binding
-	_, _, err := rbac_util.CreateOrPatchRoleBinding(c.kubeClient, meta, func(in *rbac.RoleBinding) *rbac.RoleBinding {
-		core_util.EnsureOwnerReference(&in.ObjectMeta, resource)
+	_, _, err = rbac_util.CreateOrPatchRoleBinding(c.kubeClient, meta, func(in *rbac.RoleBinding) *rbac.RoleBinding {
```
Comment (hossainemruz) [inline pkg/controller/rbac_jobs.go]: Check  `exists` first then call this function.

File: pkg/controller/rbac_jobs.go [hunk 3]
```diff
@@ -519,3 +529,51 @@ func (c *StashController) ensureBackupJobRoleBinding(resource *core.ObjectRefere
 	})
 	return err
 }
+
+func (c *StashController) checkAndGetWorrkloadReference(bcObj *api_v1beta1.BackupConfiguration) (bool bool, ref *core.ObjectReference) {
```
Comment (hossainemruz) [inline pkg/controller/rbac_jobs.go]: You also have to handle
- DeploymentConfig
- PersistentVolumeClaim
- AppBinding

File: test/e2e/framework/functions.go [hunk 4]
```diff
@@ -53,7 +53,7 @@ func (f *Invocation) UpdateStatusFunction() v1beta1.Function {
 				fmt.Sprintf("--backup-session=${%s:=}", apis.BackupSession),
 				fmt.Sprintf("--restore-session=${%s:=}", apis.RestoreSession),
 				fmt.Sprintf("--output-dir=${%s:=}", outputDir),
-				fmt.Sprintf("--enable-status-subresource=${%s:=%s}", apis.StatusSubresourceEnabled, apis.StatusSubresourceEnabled),
+				fmt.Sprintf("--enable-status-subresource=${%s:=false}", apis.StatusSubresourceEnabled),
```
Comment (hossainemruz) [inline test/e2e/framework/functions.go]: Set default value to `true`.

File: pkg/controller/rbac_jobs.go (hunk 1 above)
Comment (suaas21) [reply pkg/controller/rbac_jobs.go]: fixed

File: pkg/controller/rbac_jobs.go [hunk 5]
```diff
@@ -417,8 +419,18 @@ func (c *StashController) ensureRestoreJobRoleBinding(resource *core.ObjectRefer
...
 	}
-	_, _, err := rbac_util.CreateOrPatchRoleBinding(c.kubeClient, meta, func(in *rbac.RoleBinding) *rbac.RoleBinding {
-		core_util.EnsureOwnerReference(&in.ObjectMeta, resource)
+
+	rsObj, err := c.stashClient.StashV1beta1().RestoreSessions(resource.Namespace).Get(resource.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	exists, objRef := c.checkAndGetWorrkloadReference(rsObj.Spec.Target.Ref, rsObj.Namespace)
```
Comment (hossainemruz) [inline pkg/controller/rbac_jobs.go]: Why we need this for Restore Job? Restore Job's owner should be the Job itself.

File: pkg/controller/rbac_jobs.go [hunk 6]
```diff
@@ -500,8 +512,18 @@ func (c *StashController) ensureBackupJobRoleBinding(resource *core.ObjectRefere
...
 	}
-	_, _, err := rbac_util.CreateOrPatchRoleBinding(c.kubeClient, meta, func(in *rbac.RoleBinding) *rbac.RoleBinding {
-		core_util.EnsureOwnerReference(&in.ObjectMeta, resource)
+
+	bcObj, err := c.stashClient.StashV1beta1().BackupConfigurations(resource.Namespace).Get(resource.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	exists, objRef := c.checkAndGetWorrkloadReference(bcObj.Spec.Target.Ref, bcObj.Namespace)
```
Comment (hossainemruz) [inline pkg/controller/rbac_jobs.go]: Same here.

**Comment by suaas21** [comment]
In stash v1beta1 version, we set workload as the owner of RBAC stuff like InitContainerRolebinding and SidecarRolebinding, So the issue has been solved.

**Pull request closed without merging**